### PR TITLE
(snowpack) - component runtime check

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Now that we can conditionally accept I think this is the more ideal way to go about things. Works like a charm btw!

CC @FredKSchott

### CODE REVIEW

File: packages/snowpack/src/index.js [hunk 1]
```diff
@@ -58,14 +29,35 @@ export default function preactRefreshPlugin(config, pluginOptions) {
...
+            __module_exports__.push(type.name);
+          };
 
           ${contents}
 
           window.$RefreshSig$ = prevRefreshSig;
           window.$RefreshReg$ = prevRefreshReg;
 
-          ${hasRegister || hasJsx || isComponent(lastPart) ? postLude : ''}
+          if (__module_exports__.some(shouldPrefreshBind) && import.meta.hot) {
```
Comment (FredKSchott) [inline packages/snowpack/src/index.js:41]: super nit: `if (import.meta.hot && __module_exports__.some(shouldPrefreshBind))` makes it more likely that the reader will see the "this is HMR-related" piece of the conditional

**Comment by FredKSchott** [comment]
Awesome!!! super excited to see all of this coming together

File: packages/snowpack/src/index.js (hunk 1 above)
Comment (JoviDeCroock) [reply packages/snowpack/src/index.js:41]:
```suggestion
          if (import.meta.hot && __module_exports__.some(shouldPrefreshBind)) {
```

Good suggestion thanks

**Comment by JoviDeCroock** [comment]
@FredKSchott This has been published as a new patch version

**Pull request merged**